src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/ExternalSettingsProviderService.cs

```diff
@@ -0,0 +1,229 @@
+        private const string MonikerPackageReference = "package-reference";
+        private const string MonikerPackagesConfig = "packages-config";
+        private const string MonikerShowPackageManagementChooser = "packageManagement.showPackageManagementChooser";
+
+        private readonly ISettings _settings;
+        private readonly VSSettings _vsSettings;
+
+        private PackageRestoreConsent _packageRestoreConsent;
+        private BindingRedirectBehavior _bindingRedirectBehavior;
+        private PackageManagementFormat _packageManagementFormat;
+
+        public ExternalSettingsProviderService()
+        {
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            _settings = componentModel.GetService<ISettings>();
+            _vsSettings = _settings as VSSettings;
+            if (_vsSettings != null)
+            {
+                _vsSettings.SettingsChanged += VsSettings_SettingsChanged;
+            }
+            Debug.Assert(_settings != null);
+        }
+
+        private void VsSettings_SettingsChanged(object sender, EventArgs e)
+        {
+            _packageRestoreConsent = null;
+            _bindingRedirectBehavior = null;
+            _packageManagementFormat = null;
+            SettingValuesChanged.Invoke(this, ExternalSettingsChangedEventArgs.SomeOrAll);
+        }
+
+        private BindingRedirectBehavior BindingRedirectBehavior
+        {
+            get
+            {
+                if (_bindingRedirectBehavior is null)
+                {
+                    _bindingRedirectBehavior = new BindingRedirectBehavior(_settings);
+                }
+
+                return _bindingRedirectBehavior;
+            }
+        }
+
+        private PackageRestoreConsent PackageRestoreConsent
+        {
+            get
+            {
+                if (_packageRestoreConsent is null)
+                {
+                    _packageRestoreConsent = new PackageManagement.PackageRestoreConsent(_settings);
+                }
+
+                return _packageRestoreConsent;
+            }
+        }
+
+        private PackageManagementFormat PackageManagementFormat
+        {
+            get
+            {
+                if (_packageManagementFormat is null)
+                {
+                    _packageManagementFormat = new PackageManagementFormat(_settings);
+                }
+
+                return _packageManagementFormat;
+            }
+        }
+
+        public event EventHandler<ExternalSettingsChangedEventArgs> SettingValuesChanged;
+        public event EventHandler<EnumSettingChoicesChangedEventArgs> EnumSettingChoicesChanged { add { } remove { } }
+        public event EventHandler<DynamicMessageTextChangedEventArgs> DynamicMessageTextChanged { add { } remove { } }
+        public event EventHandler ErrorConditionResolved { add { } remove { } }
+
+        public void Dispose()
```

Comment:
IExternalSettingsProvider no longer implements IDisposable.  Instead, if you want to release resources when your external settings aren't in use, implement ISuspendableExternalSettingsProvider.

Reply:
We should talk offline, as I'm unable to find that provider. Perhaps I need a particular version to access that.

I debugged, and as you imply, I don't see the Dispose triggered when closing USX. I only see the Dispose called when closing VS and our NuGetPackage itself begins disposing resources.
![image](https://github.com/user-attachments/assets/9c204fc2-fa6b-46b0-89ae-0e3936500635)

Reply:
Chatted offline. This is available in a newer SDK. It's a perf optimization, but not a perf issue in that it will not cause any memory leaks, etc. I will update this in a separate PR after this is merged.